images/handlers.go

```diff
@@ -175,14 +175,41 @@ func SetChildrenLabels(manager content.Manager, f HandlerFunc) HandlerFunc {
 		}
 
 		if len(children) > 0 {
-			info := content.Info{
-				Digest: desc.Digest,
-				Labels: map[string]string{},
-			}
-			fields := []string{}
-			for i, ch := range children {
-				info.Labels[fmt.Sprintf("containerd.io/gc.ref.content.%d", i)] = ch.Digest.String()
-				fields = append(fields, fmt.Sprintf("labels.containerd.io/gc.ref.content.%d", i))
+			var (
+				info = content.Info{
+					Digest: desc.Digest,
+					Labels: map[string]string{},
+				}
+				fields = []string{}
+				config bool
+				layers int
+				other  int
+			)
+			for _, ch := range children {
+				var key string
+				switch ch.MediaType {
+				case MediaTypeDockerSchema2Config, ocispec.MediaTypeImageConfig:
+					if !config {
+						key = "containerd.io/gc.ref.content.config"
```

Comment:
are we need to support legacy schema 1?

Reply:
I guess for schema1 this would have to be done on conversion since the handler never sees these types. In the end though schema1 would not show up here since we never keep a schema1 manifest in the content store

Reply:
yeah. it should be done in the schema1's conversion. :) 

So this change should be also apploed to schema1's conversion.